src/main/kotlin/systems/ajax/englishstudytelegrambot/service/WordServices.kt

```diff
@@ -0,0 +1,88 @@
+package systems.ajax.englishstudytelegrambot.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.server.ResponseStatusException
+import systems.ajax.englishstudytelegrambot.dto.AudioForWordResponse
+import systems.ajax.englishstudytelegrambot.dto.DefinitionOfWordResponse
+import systems.ajax.englishstudytelegrambot.dto.ExampleOfWordResponse
+import systems.ajax.englishstudytelegrambot.dto.PronunciationOfWordResponse
+import systems.ajax.englishstudytelegrambot.dto.GettingPartOfAdditionalInfoAboutWord
+import systems.ajax.englishstudytelegrambot.entity.AdditionalInfoAboutWord
+import kotlin.reflect.KSuspendFunction1
+
+interface AdditionalInfoAboutWordService {
+    suspend fun findAdditionInfoAboutWord(wordSpelling: String): AdditionalInfoAboutWord
+}
+
+@Service
+class AdditionalInfoAboutWordServiceImpl(private val webClient: WebClient) : AdditionalInfoAboutWordService {
+
+    @Value("\${wordnik.api.key}")
+    lateinit var wordnikAPIKey: String
+
+    @Value("\${link.audio}")
+    lateinit var audioLink: String
+
+    @Value("\${link.definition}")
+    lateinit var definitionLink: String
+
+    @Value("\${link.example}")
+    lateinit var exampleLink: String
+
+    @Value("\${link.pronunciation}")
+    lateinit var pronunciationLink: String
```

Comment:
field injection.

It's better to create configuration properties class
```kotlin
@ConfigurationProperties(prefix = "link")
data class LinkProperties @ConstructorBinding constructor(
     val definition: String,
     val audioLink: String,
     ....
)

@Configuration
@EnableConfigurationProperties(LinkProperties::class)
class BotConfiguration

@Service
class AdditionalInfoAboutWordServiceImpl(
    private val webClient: WebClient,
    private val linkProperties: LinkProperties,
)
```

Reply:
these two classes LinkProperties and BotConfiguration, write in Configuration file, or above AdditionalInfoAboutWordServiceImpl ?

Reply:
would be better to create one bean with all properties (include base url and key to api), or leave three property classes(like in last commit), where each is responsible for it logic ?

Reply:
It depends. Prefer grouping by feature that it set up.

Note: `LinkProperties` may contain nested classes